### PR TITLE
Check if a hold has already been placed for an item by the same user

### PR DESCRIPTION
**Pull request by sbujam**
The check is mainly required for certain old buggy Koha versions
that allowed multiple holds from the same user to the same item
via ILS-DI.
In addition, it presents an error message to end users in case
the hold could not be placed on an item because it already existed
(even in cases where Koha ILS-DI erroneously does not return an
error code).

### CODE REVIEW

**Comment by sbujam** [comment]
NB: the code snippet was written by @OdysseasKr
Tests were made by me against Koha v3.20.10 that appeared to allow multiple holds for the same item by the same user via ILS-DI. Intermediate versions seem to disallow this behaviour, but probably(?) do not return an error. Recent versions seem to also return a 'NonHoldable' error code in ILS-DI.

**Comment by sbujam** [comment]
Maybe the returned error message could be also translatable... If you agree with the phrase a translation could also be added.

**Comment by sbujam** [comment]
Given the chance, I'm also thinking of making the errors returned to the user by the placeHold function a bit more human readable (and probably translatable). Would you prefer a separate PR for that?

**Comment by demiankatz** [comment]
@misilot, do you want to review this before I merge it? It looks fine to me but of course I can't test it from here.

@sbujam, if you want to open a separate PR with improved messages/translation support, that might be a good thing to work on after this one gets merged. I'd encourage you to examine other ILS drivers and see if there is an opportunity to unify any messaging for convenient reuse. Please let me know if you'd like any input from me on any part of the strategy!

File: module/VuFind/src/VuFind/ILS/Driver/KohaILSDI.php [hunk 1]
```diff
@@ -604,6 +604,22 @@ public function placeHold($holdDetails)
         $this->debug("Needed before date: " . $needed_before_date);
         $this->debug("Level: " . $level);
 
+        // The following check is mainly required for certain old buggy Koha versions
+        // that allowed multiple holds from the same user to the same item
+        $sql = "select count(*) as RCOUNT from reserves where borrowernumber = :rid "
+            . "and biblionumber = :bid";
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/KohaILSDI.php]: I think the main use case of concern is the one that @EreMaijala brought up on the mailing list: a bib record representing a multi-volume set, with items for different volumes. In this (fairly common) scenario, a user may indeed wish to place holds on more than one item.

Comment (sbujam) [inline module/VuFind/src/VuFind/ILS/Driver/KohaILSDI.php]: You have a point there! I didn't think of multi-volume sets. That would indeed be a case when one might want to place holds on more than one items of a title.
I'll loosen the check and recommit.

**Comment by demiankatz** [comment]
@misilot, thanks for the review -- mind taking a look at the latest revisions when time permits?

**Comment by misilot** [comment]
@demiankatz this makes sense, and I think answers my concern about requesting multiple of the same item.

**Comment by demiankatz** [comment]
Great, thanks -- merged!

**Comment by sbujam** [comment]
@demiankatz and @misilot thank you for your time and valuable feedback on this!

**Pull request merged**